### PR TITLE
[Merged by Bors] - fix: when apply? fails, say apply? not exact?

### PR DESCRIPTION
**Pull request by Julian**
(Previously the message would say `"exact? didn't find any relevant lemmas"` even when the tactic used was `apply?`)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Tactic/LibrarySearch.lean [hunk 1]
```diff
@@ -271,7 +271,8 @@ def exact? (tk : Syntax) (required : Option (Array (TSyntax `term))) (requireClo
       for suggestion in suggestions do
         withMCtx suggestion.1 do
           addExactSuggestion tk (← instantiateMVars (mkMVar mvar)).headBeta (addSubgoalsMsg := true)
-      if suggestions.isEmpty then logError "exact? didn't find any relevant lemmas"
+      let tacticName := if requireClose then "exact" else "apply"
+      if suggestions.isEmpty then logError s!"{tacticName}? didn't find any relevant lemmas"
```
Comment (alexjbest) [inline Mathlib/Tactic/LibrarySearch.lean:275]: It looks like we should just say `apply?` here, as if `requireClose` is true then an error will instead be thrown on line 269 above

Comment (Julian) [reply Mathlib/Tactic/LibrarySearch.lean:275]: Indeed, thanks.

Comment (alexjbest) [reply Mathlib/Tactic/LibrarySearch.lean]: A more flexible alternative is to use `{tk}` here which inserts the original syntax that the called passed in in quotes. But I don't think we need to worry about that right now

**Comment by alexjbest** [comment]
maintainer merge

File: Mathlib/Tactic/LibrarySearch.lean (hunk 1 above)
Comment (Julian) [reply Mathlib/Tactic/LibrarySearch.lean]: Ah interesting, good to learn regardless thanks for the tip

**Comment by Julian** [comment]
That build failure doesn't look related / looks transient. Am I able to trigger rebuilding on my own?

**Comment by RemyDegenne** [comment]
Could you please merge master then try it again?
bors d+

**Pull request closed without merging**